File: src/main/java/net/hectus/neobb/turn/default_game/TTimeLimit.java
package net.hectus.neobb.turn.default_game;

import com.marcpg.libpg.data.time.Time;
import com.marcpg.libpg.lang.Translation;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.other.OtherTurn;
import net.hectus.neobb.util.Colors;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

public class TTimeLimit extends OtherTurn<Time> implements NeutralClazz {
    public TTimeLimit(NeoPlayer player) { super(player); }
    public TTimeLimit(Time data, NeoPlayer player) { super(data, player.location(), player); }

    @Override
    public void apply() {
        player.sendMessage(Translation.component(player.locale(), "gameplay.info.too-slow.player", data.toString()).color(Colors.NEGATIVE));
        player.opponents(false).forEach(p -> p.sendMessage(Translation.component(p.locale(), "gameplay.info.too-slow." + (p == player.nextPlayer() ? "next" : "others"), player.nextPlayer()).color(Colors.NEUTRAL)));
    }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.CLOCK);
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/TTimeLimit.kt
package net.hectus.neobb.turn.default_game

import com.marcpg.libpg.data.time.Time
import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.turn.default_game.other.OtherTurn
import net.hectus.neobb.util.Colors
import net.hectus.neobb.util.component
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class TTimeLimit(data: Time?, cord: Cord?, player: NeoPlayer?) : OtherTurn<Time>(data, cord, player), NeutralClazz {
    override fun apply() {
        player!!.sendMessage(player.locale().component("gameplay.info.too-slow.player", data.toString(), color = Colors.NEGATIVE))
        player.opponents(false).forEach { p ->
            p.sendMessage(p.locale().component("gameplay.info.too-slow." + (if (p === player.nextPlayer()) "next" else "others"), player.nextPlayer().name(), color = Colors.NEUTRAL))
        }
    }

    override fun item(): ItemStack = ItemStack(Material.CLOCK)
}
